tests/test_app_centos/Dockerfile

```diff
@@ -1,4 +1,8 @@
-FROM centos:7
+FROM centos
+
+RUN cd /etc/yum.repos.d/
+RUN sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
+RUN sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*
 
 # Install Python so Ansible can run against node
 RUN yum update -y && yum install -y python3
```

Comment:
The following error occurs when upgrading centos:7 to centos:8 

`Error: Failed to download metadata for repo ‘appstream’: Cannot prepare internal mirrorlist: No URLs in mirrorlist`

According to documents, 

`After Dec 31st, 2021, if you need to update your CentOS, you need to change the mirrors to [vault.centos.org](https://vault.centos.org/) where they will be archived permanently.`

Thus, the change in this file reflects this solution.

More information regarding this issue can be found [here](https://yegorshytikov.medium.com/error-failed-to-download-metadata-for-repo-appstream-cannot-prepare-internal-mirrorlist-no-959768e5f8e5#:~:text=Fix%20Failed%20to%20download%20metadata%20for%20repo&text=that's%20means%20CentOS%208%20will,they%20will%20be%20archived%20permanently.&text=Go%20to%20the%20%2Fetc%2Fyum,repos.).